### PR TITLE
Fix inconsistent bandwidth validation for numpy scalar and non-scalar inputs

### LINKED ISSUE #92: BUG: inconsistent bandwidth validation for numpy scalar and non-scalar inputs
While testing some edge cases around fit time behaviour , i noticed few inconsistencies in how bandwidth inputs are validated.

1. Passing a numpy scalar like `np.int64(100)` raises a `ValueError: Bandwidth is not a valid value, must be int or float`, but accepts `np.float64(100.0)`, even though a normal python `int` works. This looks like a regression since both represent the numeric scalar.

2. `bandwidth=np.nan` passes validation intially but later fails inside libpysal due to missing values, instead of being rejected earlier during validation.

3. Non-scalar inputs like `[100]` or `np.array([100])` are not rejected immediately and instead fail later depending on the execution path.

All of these seem related to bandwidth validation rather than modelling logic. My understanding is that bandwidth should accept numeric scalar values only, while rejecting non-finite or non-scalar inputs early during validation.

### PR DESCRIPTION
**Pull request by AshishRaj97**
This PR handles the bandwidth validation inconsistencies as reported in #92 

changes:
- accept numeric scalar values, including numpy scalar types
- rejects non scalar inputs (like `[100]` or `np.array([100])`) during fit time validation
- reject `np.nan` bandwidth values early before reaching downstream
- allow infinite bandwidth only for fixed kernels (`fixed=True`)
- enforce interger value bandwidth for adaptive kernels (`fixed=False`)
- added internal normalization using `_bandwidth` so that estimator parameters are not mutated during fit
- removed bandwidth checks from `_build_weights` , keeping validation separate

Added tests covering the reported bandwidth validation edge cases

Closes #92

### CODE REVIEW

File: gwlearn/base.py [hunk 1]
```diff
@@ -513,12 +508,47 @@ def _validate_fit_inputs(
...
+            if bw <= 0:
+                raise ValueError("bandwidth must be a positive number.")
+            
+            # adaptive bandwidth must be finite
+            # infinite bandwidth is allowed only for fixed kernels,
+            # representing a global neighbourhood
+            if not self.fixed and not np.isfinite(bw):
+                raise ValueError(
+                    "Adaptive bandwidth must be finite."
+                )
```
Comment (martinfleis) [inline gwlearn/base.py]: Remove

**Comment by AshishRaj97** [comment]
while working on bandwidth validation, i notice that `_prepare_prediction_neighborhoods` accepts overiding bandiwdth but it validates `int or float` checks. Values like `None`, `negative` , `floats (non-integer adaptive bandwidth) ` are failing in downstream. 

so would it be better to extract bandwidth validation in shared helper and reuse it here so prediction override bandwidth follow the same validation logic or would you prefer keeping validation localized and just aligining the type checks.

File: gwlearn/base.py [hunk 2]
```diff
@@ -131,12 +131,7 @@ def _validate_geometry(self, geometry):
 
     def _build_weights(self) -> graph.Graph:
         """Build spatial weights graph"""
-        if not isinstance(self.bandwidth, float | int):
-            raise ValueError(
-                "Bandwidth is not a valid value. Needs to be float or int, "
-                f"got {self.bandwidth}."
-            )
-
+        assert self.bandwidth is not None
```
[the remark this answers is not in the document]
Comment (AshishRaj97) [reply gwlearn/base.py]: okay, i will remove the assertion
